procore.gemspec

```diff
@@ -38,6 +38,6 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "webmock"
 
   spec.add_dependency "activesupport", "> 2.4"
-  spec.add_dependency "oauth2", "~> 1.4"
-  spec.add_dependency "rest-client", "~> 2.0.0"
+  spec.add_dependency "oauth2", "~> 2.0"
```

Comment:
What do you think of relaxing the requirement to just `>= 1.4`?
- I think many Ruby apps are still using `1.x` version of `oauth2` (our apps for example `procore/procore` or `web-portal`).
- If we use `~> 2.0`, when people try to use our up-versioned `procore` gem, there could be be surprising gem version conflicts.

Reply:
I like this idea, if it will still work

Reply:
We could do that, since `1.4` doesn't cause a problem currently, but do we want to?

* Future changes to the SDK will need to be tested against both major versions of `oauth2` and we'll be restricted to only what they both support.
* Clients such as `procore/procore` should switch anyway since `1.4` is EOL, but they won't do so automatically if it's still supported. Nothing else in that repo depends on the `oauth2` gem.
* We don't want to indicate support for a future `oauth2` `3.x` version which might contain a breaking change.

The `oauth2` gem itself suggests what I have here:
```
spec.add_dependency 'oauth2', '~> 2.0'
```
https://gitlab.com/oauth-xx/oauth2/#versioning

Reply:
Cool, I am good with the suggested `~> 2.0`.